### ISSUE TITLE
Can I replicate the XNLI results with the provided MLM and TLM model

### ISSUE THREAD

**Issue by luyich** [facebookresearch/XLM#110]
Thanks for release the code!

I'm trying to replicate the result for XNLI reported in your paper using both the MLM and TLM model provided in this repo. However, I'm only able to get a average accuracy of 70.4 for MLM and 73.5 for TLM while the paper reported 71.5 and 75.1 respectively. I'm using the command provided in the repo for XNLI fine tuning.

Is the released pretrained MLM and TLM weaker than what is used in the paper or I'm not using the right hyper parameters (If so can you kindly provide what's the hyper parameter that provides the result in the paper)?

**Comment by aconneau** [comment]
Hi @luyich,

thanks for your interest in our work. For XNLI results or GLUE results, we sweep over parameters for finetuning and select the best for each language.
Typically you should try:

"optimizer": ["adam,lr=0.00001", "adam,lr=0.000005", "adam,lr=0.000002"],
and
"batch_size": [8, 16]

This should improve your results.
Best,
Alexis

**Comment by thespectrewithin** [comment]
Hi, may I ask how many epochs did you run finetuning for? I know that 250 is given in the repo but I ran the codes and found that the model would've converged much earlier. Thanks.

**Comment by aconneau** [comment]
@thespectrewithin : indeed you don't need 250 epochs. We're used to watching the results of our jobs regularly so in general we simply specify a high number of epochs and kill jobs manually when it has converged. But you can definitely decrease the number of epochs if you'd like.

**Comment by thespectrewithin** [comment]
Thanks!

**Comment by luyich** [comment]
Thank you very much!

**Comment by maksym-del** [comment]
Hi @aconneau ! 

> For XNLI results or GLUE results, we sweep over parameters for finetuning and select the best for each language.

do I understand correctly that you end up with several checkpoints resulted from finetuning with several hparams configurations and use several of them for evaluation (and not just one for all the languages)?
